### PR TITLE
Ensure that .map doesn't modify options passed in

### PR DESCRIPTION
**Pull request by ajlai**
Fixes https://github.com/grosser/parallel/issues/226

### CODE REVIEW

**Comment by ajlai** [comment]
also fixes https://github.com/grosser/parallel/issues/197

File: lib/parallel.rb [hunk 1]
```diff
@@ -231,7 +231,8 @@ def each_with_index(array, options={}, &block)
       each(array, options.merge(:with_index => true), &block)
     end
 
-    def map(source, options = {}, &block)
+    def map(source, _options = {}, &block)
+      options = _options.dup
```
Comment (grosser) [inline lib/parallel.rb]: `_` prefix means that it's not being used prefer `options = options.dup`

File: spec/parallel_spec.rb [hunk 2]
```diff
@@ -233,7 +237,7 @@ def cpus
       `ruby spec/cases/parallel_map_complex_objects.rb`.should == "YES"
     end
 
-    it "starts new process imediatly when old exists" do
+    it "starts new process immediately when old exists" do
```
Comment (grosser) [inline spec/parallel_spec.rb]: oh boy that spelling :trollface:

**Comment by grosser** [comment]
clean fix, neat test ❤️

**Comment by grosser** [comment]
yeah, needs to die :)

**Comment by grosser** [comment]
v1.13.0

**Comment by Annih** [comment]
Thanks @ajlai & @grosser

**Pull request merged**